engine/gamesys/proto/gamesys/physics_ddf.proto

```diff
@@ -340,3 +342,245 @@ message EnableGridShapeLayer
+ *
+ * `normal`
+ * : [type:vector3] normal in world space of the contact point, which points from B object towards A object
+ *
+ * @param b [type:table] contact point information for object B
+ *
+ * `position`
+ * : [type:vector3] The world position of object B.
+ *
+ * `id`
+ * : [type:hash] The ID of object B.
+ *
+ * `group`
+ * : [type:hash] The collision group of object B (hash).
+ *
+ * `relative_velocity`
+ * : [type:vector3] The relative velocity of the collision object B as observed from A object
+ *
+ * `mass`
+ * : [type:number] The mass of the collision object B in kg
+ *
+ * `normal`
+ * : [type:vector3] normal in world space of the contact point, which points from A object towards  object
+ *
+ * @examples
+ *
+ * How to take action when a contact point occurs:
+ *
+ * ```lua
+ * physics.set_listener(function(self, event, data)
+ *  if event == hash("contact_point_event") then
+ *      pprint(data)
+ *      -- {
+ *      --  applied_impulse = 310.00769042969,
+ *      --  distance = 0.0714111328125,
+ *      --  a = {
+ *      --      position = vmath.vector3(446, 371, 0),
+ *      --      relative_velocity = vmath.vector3(1.1722083854693e-06, -20.667181015015, -0),
+ *      --      mass = 0,
+ *      --      group = hash: [default],
+ *      --      id = hash: [/flat],
+ *      --      normal = vmath.vector3(-0, -1, -0)
+ *      --  },
+ *      --  b = {
+ *      --      position = vmath.vector3(185, 657.92858886719, 0),
+ *      --      relative_velocity = vmath.vector3(-1.1722083854693e-06, 20.667181015015, 0),
+ *      --      mass = 10,
+ *      --      group = hash: [default],
+ *      --      id = hash: [/go2],
+ *      --      normal = vmath.vector3(0, 1, 0)
+ *      --  },
+ *      -- }
+ *   end
+ * end)
+ * ```
+ */
+message ContactPointEvent
+{   
+    required ContactPoint   a                         = 1;
+    required ContactPoint   b                         = 2;
+    required float          distance                  = 3;
+    required float          applied_impulse           = 4;
+}
+
+message Collision
+{
+    required dmMath.Point3 position                  = 1;
+    required uint64 id                               = 2;
+    required uint64 group                            = 3;
+}
+
+/*# reports a collision between two collision objects in cases where a listener is specified.
+ * See [ref:physics.set_listener].
+ *
+ * This message is sent to a function specified in [ref:physics.set_listener]
+ * when two collision objects collide.
+ *
+ * This message only reports that a collision has occurred and will be sent once per frame.
+ * For more detailed information, check for the [ref:contact_point_event].
+ *
+ * @message
+ * @name collision_event
+ * @param a [type:table] collision information for object A
+ *
+ * `position`
+ * : [type:vector3] The world position of object A.
+ *
+ * `id`
+ * : [type:hash] The ID of object A.
+ *
+ * `group`
+ * : [type:hash] The collision group of object A (hash).
```

Comment:
No need to put (hash) at the end when it is annotated with a type using `[type:hash]`?